tests/sqllogictests/suites/base/09_fuse_engine/09_0042_merge_into_issue_16588.test

```diff
@@ -0,0 +1,24 @@
+statement ok
+create or replace database i16588;
+
+statement ok
+use i16588;
+
+statement ok
+create table base(a int);
+
+statement ok
+create table sink(a int);
+
+query II
+merge into sink using base on 1 != 1 when matched then update * when not matched then insert *;
```

Comment:
It would be better to add `explain` test for this sql.

Reply:
> It would be better to add `explain` test for this sql.

Thanks, Done!